composer/models/huggingface.py

```diff
@@ -84,10 +94,41 @@ def __init__(self,
 
         super().__init__()
         self.model = model
-        self.config = model.config
+        self.config: PretrainedConfig = model.config
         self.model_forward_args = inspect.getfullargspec(self.model.forward).args
+
+        if _peft_installed and self.model_forward_args == ['self']:
+            from peft import PeftModel
+            if isinstance(self.model, PeftModel):
+                self.model_forward_args = inspect.getfullargspec(self.model.base_model.model.forward).args
+
+        # inspect.getfullargspec HuggingFace quantized model could not return args correctly
```

Comment:
does this comment need to be here?

Reply:
maybe? This was copy pasted over from foundry. I should check whether we can unify to just use this version of the arg getting instead of the other version.